llvm/lib/Target/AMDGPU/SIISelLowering.cpp

```diff
@@ -16062,7 +16062,8 @@ SITargetLowering::getRegForInlineAsmConstraint(const TargetRegisterInfo *TRI_,
     case 'v':
       switch (BitWidth) {
```

Comment:
`switch` seems overkill here. Could just handle it with:
```
  if (BitWidth == 16 && !Subtarget->useRealTrue16Insts())
    BitWidth = 32;
```

Reply:
Or even move the handling for the `BitWidth == 16` case inside `getVGPRClassForBitWidth`?

Reply:
This reminds me we need constraints for the aligned and unaligned versions of register classes